File: android-app/app/src/main/java/app/calendariociclismo/android/data/model/Broadcast.kt
package app.calendariociclismo.android.data.model

import app.calendariociclismo.android.util.DateFormatting
import kotlinx.serialization.Serializable

/** Emisión de TV/streaming de una etapa (tabla `broadcasts`). */
@Serializable
data class Broadcast(
    val id: String,
    val raceDayId: String,
    val channel: String? = null,
    val startTimeUtc: String? = null,
    val url: String? = null,
    val note: String? = null,
    val sortOrder: Int = 0,
    val showInRevive: Boolean = false,
    val country: String? = null,
) {
    /** Hora de inicio formateada en la zona horaria del dispositivo (HH:mm). */
    val startTimeLocal: String?
        get() = startTimeUtc?.let { DateFormatting.formatTimeLocal(it) }
}

File: ios-app/CalendarioCiclismo/Models/Broadcast.swift
import Foundation

/// Emisión de TV/streaming de una etapa (tabla `broadcasts` en Supabase).
struct Broadcast: Codable, Identifiable, Hashable {
    let id: String
    let raceDayId: String
    let channel: String?
    let startTimeUtc: String?
    let url: String?
    let note: String?
    let sortOrder: Int?
    let showInRevive: Bool?
    let country: String?

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    /// Hora de inicio formateada en la zona horaria del dispositivo.
    var startTimeLocal: String? {
        guard let ts = startTimeUtc else { return nil }
        return DateFormatting.formatTimeLocal(ts)
    }
}
